Sources/Orbit/WIP/PasswordStrengthIndicator.swift

```diff
@@ -0,0 +1,107 @@
+import SwiftUI
+
+public enum PasswordStrength {
+    case empty
+    case weak(title: String)
+    case medium(title: String)
+    case strong(title: String)
+}
+
+struct PasswordStrengthIndicator: View {
+
+    let passwordStrength: PasswordStrength
+    let textAccessibilityIdentifier: String
+
+    var body: some View {
+        if text.isEmpty == false {
+            HStack(spacing: .xSmall) {
+                indicator
+
+                Text(text, size: .small, color: .custom(uiColor))
+                    .frame(maxWidth: .infinity, alignment: .trailing)
+                    .accessibility(identifier: textAccessibilityIdentifier)
+            }
+        }
+    }
+
+    var indicator: some View {
+        Capsule()
+            .frame(height: 2)
+            .foregroundColor(Color.cloudDark)
+            .overlay(
+                GeometryReader { geometry in
```

Comment:
Now that the design got more constant, we can get rid of geometry reader and calculations.

![image](https://user-images.githubusercontent.com/35844477/161230147-45a98db1-6f9a-4b4f-829f-eabd36d3311f.png)

Code:

```
import SwiftUI
import Orbit

struct ProgressView: View {

    enum Progress {
        case empty
        case weak
        case medium
        case strong
    }

    let progress: Progress

    var body: some View {
        HStack(spacing: 0) {
            Capsule()
                .fill(Color.cloudNormal)
                .overlay(progressBar)
                .frame(height: .xxSmall)
                .frame(maxWidth: .infinity, alignment: .leading)

            Text(text, color: .custom(color))
                .frame(maxWidth: .infinity, alignment: .trailing)
        }
    }

    @ViewBuilder var progressBar: some View {
        HStack(spacing: 0) {
            Capsule()
                .fill(Color(color))
                .frame(maxWidth: .infinity)

            ForEach(0 ..< spacersCount, id: \.self) { _ in
                Spacer()
                    .frame(maxWidth: .infinity)
            }
        }
    }

    var spacersCount: Int {
        switch progress {
            case .empty:        return 4
            case .weak:         return 3
            case .medium:       return 1
            case .strong:       return 0
        }
    }

    var color: UIColor {
        switch progress {
            case .empty:        return .clear
            case .weak:         return .redNormal
            case .medium:       return .orangeNormal
            case .strong:       return .greenNormal
        }
    }

    var text: String {
        switch progress {
            case .empty:        return "-"
            case .weak:         return "weak"
            case .medium:       return "medium"
            case .strong:       return "strong"
        }
    }
}

struct ProgressViewPreviews: PreviewProvider {
    
    static var previews: some View {
        PreviewWrapper {
            ProgressView(progress: .empty)
            ProgressView(progress: .weak)
            ProgressView(progress: .medium)
            ProgressView(progress: .strong)
        }
        .previewLayout(.sizeThatFits)
    }
}

```

Reply:
Applied suggestion with one change — the whole component is hidden if the value is empty, no need for "-" text in that case according to the design.